cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3896,6 +3896,41 @@ INSTANTIATE_TEST_SUITE_P(WriteDictionary, TestArrowWriteDictionary,
                          ::testing::Values(ParquetDataPageVersion::V1,
                                            ParquetDataPageVersion::V2));
 
+TEST(TestArrowWriteNestedDictionary, TestNullCount) {
+  std::shared_ptr<::arrow::Array> dictionary = ArrayFromJSON(::arrow::utf8(), R"(["z"])");
+  std::shared_ptr<::arrow::Array> indices =
+      ArrayFromJSON(::arrow::int32(), R"([0, null, 0])");
+  std::shared_ptr<::arrow::Array> offsets =
+      ArrayFromJSON(::arrow::int32(), R"([null, 0, null, null, null, 3])");
+  ASSERT_OK_AND_ASSIGN(std::shared_ptr<::arrow::Array> dict_encoded,
+                       ::arrow::DictionaryArray::FromArrays(indices, dictionary));
+  // values looks like [null, ["z", null, "z"], null, null, null]
+  ASSERT_OK_AND_ASSIGN(
+      std::shared_ptr<Array> values,
+      ::arrow::ListArray::FromArrays(*offsets, *dict_encoded, default_memory_pool()));
+  std::shared_ptr<::arrow::Schema> schema =
+      ::arrow::schema({::arrow::field("values", values->type())});
+  std::shared_ptr<::arrow::Table> table = ::arrow::Table::Make(schema, {values});
+
+  std::shared_ptr<::arrow::ResizableBuffer> serialized_data = AllocateBuffer();
+  auto out_stream = std::make_shared<::arrow::io::BufferOutputStream>(serialized_data);
+  std::unique_ptr<FileWriter> writer;
+  ASSERT_OK(FileWriter::Open(*table->schema(), default_memory_pool(), out_stream,
+                             default_writer_properties(),
+                             default_arrow_writer_properties(), &writer));
+  ASSERT_OK(writer->WriteTable(*table, std::numeric_limits<int64_t>::max()));
+  ASSERT_OK(writer->Close());
+  if (!out_stream->closed()) {
```

Comment:
It is probably not necessary to test for `out_stream->closed()`, is it?

Reply:
made it unconditional.